irif/inc/irif.h

```diff
@@ -27,6 +27,8 @@ extern __attribute__((const)) half2 __llvm_trunc_2f16(half2) __asm("llvm.trunc.v
 extern __attribute__((const)) half2 __llvm_round_2f16(half2) __asm("llvm.round.v2f16");
 extern __attribute__((const)) half2 __llvm_rint_2f16(half2) __asm("llvm.rint.v2f16");
 extern __attribute__((const)) half2 __llvm_canonicalize_2f16(half2) __asm("llvm.canonicalize.v2f16");
+extern __attribute__((const)) int __llvm_ctpop_i32(int) __asm("llvm.ctpop.i32");
```

Comment:
Thanks for the point. These used in cuda2gcn library, I built for the Triton project.
There are indeed builtins `__builtin_popcount()` and `__builtin_popcountl()`. Also, there are functions in ockl library.

So what is the best approach in cuda2gcn:
1. call explicitly `__builtin_popcount..`, or
2. call ockl with using e.g. OCKL_MANGLE_U32 macro?

Reply:
I think triton is calling LLVM builtins freely (not sure that's a great idea) so might as well call the popcount builtin as well.

Reply:
Actually I asked about cuda2gcn library in this repository (it is not build by default). Perhaps we need an option to build it?

Basically triton uses __nv_* library functions. So to support triton on AMDGPU we need to port or wrap them and I found cuda2gcn in this repository that already has these wrappers.

Reply:
Are we  OK with triton doing this?  Why is triton overspecializing for one architecture?  Seems like it would be better to fix at the root than trying to pretend that all GPUs look like nvidia's.

Reply:
Originally, triton project was unaware of other targets than NV. AMD target supported there by side developers.
I agree, it would be better to refactor the code. But at the moment triton project is in pre-release state and using the wrappers seems best solution there at the moment.

@whchung what do you think?